metricbeat/module/system/process/helper.go

```diff
@@ -363,16 +368,72 @@ func (procStats *ProcStats) GetProcStats() ([]common.MapStr, error) {
 			}
 
 			newProcs[process.Pid] = process
-
 			last := procStats.ProcsMap[process.Pid]
-			proc := procStats.GetProcessEvent(process, last)
+			process.cpuTotalPct = GetProcCpuPercentage(last, process)
+			processes = append(processes, *process)
+		}
+	}
+	procStats.ProcsMap = newProcs
+
+	if procStats.IncludeTop.Enabled {
+		processes = procStats.includeTopProcesses(processes)
+		logp.Debug("processes", "Filtered top processes down to %d processes", len(processes))
+	}
+
+	procs := []common.MapStr{}
+	for _, process := range processes {
+		proc := procStats.GetProcessEvent(&process)
+		procs = append(procs, proc)
+	}
+
+	return procs, nil
+}
+
+func (p *ProcStats) includeTopProcesses(processes []Process) []Process {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
receiver name p should be consistent with previous receiver name procStats for ProcStats